### ISSUE TITLE
log --first-parent is missing?

### ISSUE THREAD

**Issue by tenpn** [magit/magit#2012]
I use log --first-parent a lot to look at recent features merged into master. In the previous version, this was accessible via `l - f l`. In magit 2.1, it seems gone from the log options window.

Any way to make it re-appear? Or maybe there's another way to do what I want?

**Comment by tarsius** [comment]
You can add new arguments, see http://magit.vc/manual/magit-popup/Customizing-existing-popups.html.

**Comment by tenpn** [comment]
Awesome! Although this is giving me an error:

    (magit-define-popup-switch magit-log-popup ?f "first parent" "--first-parent")

wrong type argument error:

    Debugger entered--Lisp error: (wrong-type-argument symbolp (:variable magit-log-arguments :man-page "git-log" :switches ((103 "Show graph" "--graph") (99 "Show graph in color" "--color") (100 "Show refnames" "--decorate") (83 "Show signatures" "--show-signature") (117 "Show diffs" "--patch") (115 "Show diffstats" "--stat") (68 "Simplify by decoration" "--simplify-by-decoration")) :options ((102 "Limit to files" "-- " magit-read-files) (97 "Limit to author" "--author=" read-from-minibuffer) (109 "Search messages" "--grep=" read-from-minibuffer) (112 "Search patches" "-G" read-from-minibuffer)) :actions ((108 "Log current" magit-log-current) (76 "Log local branches" magit-log-branches) (114 "Reflog current" magit-reflog-current) (111 "Log other" magit-log) (98 "Log all branches" magit-log-all-branches) (79 "Reflog other" magit-reflog) (104 "Log HEAD" magit-log-head) (97 "Log all references" magit-log-all) (72 "Reflog HEAD" magit-reflog-head)) :default-arguments ("--graph" "--decorate") :default-action magit-log-current :max-action-columns 3))
      magit-define-popup-key((:variable magit-log-arguments :man-page "git-log" :switches ((103 "Show graph" "--graph") (99 "Show graph in color" "--color") (100 "Show refnames" "--decorate") (83 "Show signatures" "--show-signature") (117 "Show diffs" "--patch") (115 "Show diffstats" "--stat") (68 "Simplify by decoration" "--simplify-by-decoration")) :options ((102 "Limit to files" "-- " magit-read-files) (97 "Limit to author" "--author=" read-from-minibuffer) (109 "Search messages" "--grep=" read-from-minibuffer) (112 "Search patches" "-G" read-from-minibuffer)) :actions ((108 "Log current" magit-log-current) (76 "Log local branches" magit-log-branches) (114 "Reflog current" magit-reflog-current) (111 "Log other" magit-log) (98 "Log all branches" magit-log-all-branches) (79 "Reflog other" magit-reflog) (104 "Log HEAD" magit-log-head) (97 "Log all references" magit-log-all) (72 "Reflog HEAD" magit-reflog-head)) :default-arguments ("--graph" "--decorate") :default-action magit-log-current :max-action-columns 3) :switches 102 ("first parent" "--first-parent" nil) nil nil)
      magit-define-popup-switch((:variable magit-log-arguments :man-page "git-log" :switches ((103 "Show graph" "--graph") (99 "Show graph in color" "--color") (100 "Show refnames" "--decorate") (83 "Show signatures" "--show-signature") (117 "Show diffs" "--patch") (115 "Show diffstats" "--stat") (68 "Simplify by decoration" "--simplify-by-decoration")) :options ((102 "Limit to files" "-- " magit-read-files) (97 "Limit to author" "--author=" read-from-minibuffer) (109 "Search messages" "--grep=" read-from-minibuffer) (112 "Search patches" "-G" read-from-minibuffer)) :actions ((108 "Log current" magit-log-current) (76 "Log local branches" magit-log-branches) (114 "Reflog current" magit-reflog-current) (111 "Log other" magit-log) (98 "Log all branches" magit-log-all-branches) (79 "Reflog other" magit-reflog) (104 "Log HEAD" magit-log-head) (97 "Log all references" magit-log-all) (72 "Reflog HEAD" magit-reflog-head)) :default-arguments ("--graph" "--decorate") :default-action magit-log-current :max-action-columns 3) 102 "first parent" "--first-parent")
      eval((magit-define-popup-switch magit-log-popup 102 "first parent" "--first-parent") nil)
      eval-last-sexp-1(nil)
      eval-last-sexp(nil)
      call-interactively(eval-last-sexp nil nil)
      command-execute(eval-last-sexp)

**Comment by bastianbeischer** [comment]
Replace magit-log-popup with 'magit-log-popup and it should work.

**Comment by tenpn** [comment]
:+1: all working!

**Comment by theldoria** [comment]
How would you do the same with the new transient interface?

How can I define it so that I can support both variants (since my config is used on several machines, some in which I update elpa content only if really required).

**Comment by tarsius** [comment]
> How would you do the same with the new transient interface?

https://magit.vc/manual/transient/Enabling-and-Disabling-Suffixes.html

> How can I define it so that I can support both variants 

```emacs-lisp
(if (require 'transient nil t)
    THEN
  ELSE)
```

**Comment by aesyondu** [comment]
Hello, as instructed here, https://github.com/magit/magit/wiki/Converting-popup-modifications-to-transient-modifications#transient-2, I pasted this in my init.el
```
(transient-append-suffix 'magit-log "-A"
			 '("-1" "First parent" "--first-parent"))
```

However, I'm getting `Symbol's function definition is void: transient-append-suffix`

I'm not sure what to do, I'm still a beginner with emacs in general, moreso with magit.

**Comment by npostavs** [comment]
@aesyondu You need to run that expression only after magit is loaded:

```elisp
(with-eval-after-load 'magit
  (transient-append-suffix 'magit-log "-A"
			   '("-1" "First parent" "--first-parent")))
```

**Comment by aesyondu** [comment]
> ```emacs-lisp
> (with-eval-after-load 'magit
>   (transient-append-suffix 'magit-log "-A"
> 			   '("-1" "First parent" "--first-parent")))
> ```

Very nice, it works. Thank you!

**Comment by MatthewFluet** [comment]
But, with current Magit, there is a `--first-parent` suffix for `magit-log`, just not enabled by default; see https://github.com/magit/magit/blob/master/lisp/magit-log.el#L417.

Certain transient popups hide some of their suffix commands by default. These additional suffixes can be [enabled interactively](https://magit.vc/manual/transient/Enabling-and-Disabling-Suffixes.html). TL;DR Enter the transient as usual and type `C-x l`.

**Comment by aesyondu** [comment]
> But, with current Magit, there is a `--first-parent` suffix for `magit-log`, just not enabled by default; see https://github.com/magit/magit/blob/master/lisp/magit-log.el#L417.
> 
> Certain transient popups hide some of their suffix commands by default. These additional suffixes can be [enabled interactively](https://magit.vc/manual/transient/Enabling-and-Disabling-Suffixes.html). TL;DR Enter the transient as usual and type `C-x l`.

Nice! After showing the log popup, I pressed `C-x l`, then pressed `=p`, `1`, then `RET`. I don't really understand what the green 1s and the red 6 and 7s mean. But it works and it persists after I close emacs.
